test/authentication/user/middleware/index.spec.unit.ts

```diff
@@ -28,7 +28,14 @@ describe("Authentication middleware", () => {
   it("should redirect to start page if loading start page with trailing slash", async () => {
     const response = await request(app)
       .get("/promise-to-file/");
-    expect(response.status).toEqual(200);
+    expect(response.status).toEqual(302);
+  });
+
+  it("should redirect to start page if /promise-to-file/* called and signed in", async () => {
+    const response = await request(app)
+      .get("/promise-to-file/company-number")
+      .expect("Location", "/promise-to-file");
```

Comment:
I think you should populate the referer here, it looks liek it is redirecting for the wrong reason ie. referer is blank

Reply:
Changed the way the unit tests are designed to better test the new functionality